### PR TITLE
DMP-2187: Admin portal routing

### PR DESCRIPTION
**Pull request by deanbullock-hmcts**
### Jira link (if applicable)

[DMP-2187](https://tools.hmcts.net/jira/browse/DMP-2187)

### Change description ###

- Split Admin and Portal routes
- Placeholder User component
- Hide nav on 404
- Update tests
- Add admin nav
- Add stub ADMIN user

### CODE REVIEW

File: darts-api-stub/users.js [hunk 1]
```diff
@@ -59,9 +59,18 @@ module.exports = [
...
     code: 'judge-approver',
     userState: {
-      userId: 2,
+      userId: 7,
       userName: 'martin.adams@darts.local',
       roles: [JUDGE, APPROVER],
     },
   },
+  {
+    name: 'Luke Littler',
```
Comment (davet1985) [inline darts-api-stub/users.js:68]: 🔥 🎯

File: src/app/admin/admin.routes.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// admin.routes.ts
```
Comment (davet1985) [inline src/app/admin/admin.routes.ts:1]: Not sure we need this comment?

File: src/app/components/layout/header/header.component.html [hunk 3]
```diff
@@ -41,81 +41,159 @@
...
-            <a
-              class="moj-primary-navigation__link"
-              routerLink="search"
-              routerLinkActive="active"
-              ariaCurrentWhenActive="page"
-              >Search</a
-            >
-          </li>
-          @if (userService.isTranscriber()) {
+        @if (userService.isAdmin() && router.url.startsWith('/admin')) {
```
Comment (davet1985) [inline src/app/components/layout/header/header.component.html:44]: Is it worth splitting this out into `portal-header` and `admin-header` components?

**Review by bpires2 — CHANGES_REQUESTED** [review]

File: src/app/components/layout/header/header.component.html (hunk 3 above)
Comment (deanbullock-hmcts) [reply src/app/components/layout/header/header.component.html:44]: 100% Yes - I think it should be done in another PR / ticket though. I tried to keep this PR from getting too big and get it in asap so we can get cracking with the rest of the admin screens.

File: src/app/auth/auth.guard.ts [hunk 4]
```diff
@@ -35,6 +35,12 @@ export const authGuard: CanActivateFn = (route: ActivatedRouteSnapshot, state: R
           if (userService.hasRoles(route.data.allowedRoles)) {
             return true;
           } else {
+            if (route.data.allowedRoles.includes('ADMIN') && route.url[0].path === 'admin') {
```
[the remark this answers is not in the document]
Comment (deanbullock-hmcts) [reply src/app/auth/auth.guard.ts]: It seems to be, I've moved it out to simplify it a bit.

**Review by bpires2 — APPROVED** [review]

**Pull request merged**